app/invocation/invocation_raw_logs_card.tsx

```diff
@@ -32,6 +32,10 @@ export default class RawLogsCardComponent extends React.Component {
     this.setState({ ...this.state, numPages: this.state.numPages + 1 });
   }
 
+  handleAllClicked() {
+    this.setState({ ...this.state, numPages: Number.MAX_SAFE_INTEGER });
```

Comment:
This is probably non-obvious, but writing `this.setState({ ...this.state, foo: bar })` is redundant because `setState` (somewhat confusingly) updates only the provided state keys, and any keys which are not present in the passed in state updates object are preserved from the existing state.

So you can remove the `...this.state` and just write `this.setState({ numPages: Number.MAX_SAFE_INTEGER })`. Bonus points if you would like to clean up the other places in this file where we do this as well -- `this.setState(this.state)` in `handleEventClicked` can also be replaced with `this.setState({})`.

Reply:
Thanks; done!